cg/models/cg_config.py

```diff
@@ -458,12 +462,29 @@ def pdc_api(self) -> PdcAPI:
             self.pdc_api_ = api
         return api
 
+    @property
+    def slurm_service(self) -> SlurmService:
+        return SlurmCLIService()
+
+    @property
+    def slurm_upload_service(self) -> SlurmUploadService:
+        slurm_upload_config = SlurmUploadConfig(
+            email=self.data_delivery.mail_user,
+            account=self.data_delivery.account,
+            log_dir=self.data_delivery.base_path,
+        )
+        return SlurmUploadService(
+            slurm_service=self.slurm_service,
+            trailblazer_api=self.trailblazer_api,
+            config=slurm_upload_config,
+        )
+
     @property
     def scout_api(self) -> ScoutAPI:
         api = self.__dict__.get("scout_api_")
         if api is None:
             LOG.debug("Instantiating scout api")
-            api = ScoutAPI(config=self.dict())
+            api = ScoutAPI(config=self.dict(), slurm_upload_service=self.slurm_upload_service)
```

Comment:
Can we not also define this in the `CGConfig(BaseModel)` above? Then we avoid needing to pass it on like this and can just get it from the config like the the `HousekeeperAPI` for instance.

Reply:
Do you mean that `ScoutAPI` should access the `slurm_upload_service` via the config instead?
I think it is better to be explicit when composing objects

Reply:
Leaving it as is for now since it is only used in the scout api

Reply:
Indeed, via the config as this is already passed anyway, I don't really see the benefit of passing it as an extra argument when we already get use the config during instantiation of the api